fidesctl/src/fidesctl/cli/utils.py

```diff
@@ -8,13 +8,31 @@
 from typing import Any, Callable, Dict
 
 import click
+import requests
 from fideslog.sdk.python.event import AnalyticsEvent
 from fideslog.sdk.python.exceptions import AnalyticsException
 from fideslog.sdk.python.utils import OPT_OUT_COPY
-from requests import Response
 
 from fidesctl.core.config.utils import get_config_from_file, update_config_file
-from fidesctl.core.utils import echo_red
+from fidesctl.core.utils import echo_red, check_response
+from fidesctl.core import api as _api
+
+
+def check_server(cli_version: str, server_url: str) -> None:
+    "Runs a health check and a version check against the server."
+    healthcheck_url = server_url + "/health"
+    version_url = server_url + "/version"
+    try:
+        check_response(_api.ping(healthcheck_url))
+    except requests.exceptions.ConnectionError as err:
+        echo_red("Connection failed, webserver is unreachable.")
+        raise SystemExit(err)
```

Comment:
Is it more appropriate to simply exit with a 1 status here? Why raise an exception if we're going to log the error in red as well?

Reply:
> the `err` in this case shows users exactly what the URL being used was

I agree it's helpful for the user to have details about the error, but it's less helpful for a user to see a Python stack trace with potentially unexpected information just because they happen to be using fidesctl. Re: my previous comments, I think the most straightforward way to handle this scenario is to `raise` the `ConnectionError`, and catch it in the `status` function, so we can create a helpful/friendly error message without leaking implementation details.

> this can be very helpful when debugging especially if something is getting configured somewhere that they don't realize.
>
> They might think their file is being used and confused why its not working, when in reality there is an env var set that they forgot about. This would surface that.

We have [other work to surface the API config values](https://github.com/ethyca/fides/issues/326) that are in use. Maybe we create a new issue to do something similar for the CLI?

Reply:
good call here, I'll update the messages accordingly and make sure that both fail